### ISSUE TITLE
[BUG]: Incorrect Version After 5.1.1 Update 

### ISSUE THREAD

**Issue by kgrammer** [phalcon/cphalcon#16212]
After updating to Phalcon 5.1.1, the reported version using the follow code returns "5.1.0"...

        $phalcon_version = new \Phalcon\Support\Version();
        $dashboard['phalcon_version'] = $phalcon_version->get();                    # Get the version of Phalcon we are running

I used the compiled method of installing Phalcon 5.1.1 on about a half dozen Phalcon 5 sites.

**Comment by Jeckerson** [comment]
Just tried:
```bash
pecl install phalcon
php --ri phalcon
```

Output:
```
phalcon


Phalcon is a full stack PHP framework, delivered as a PHP extension, offering lower resource consumption and high performance.
phalcon => enabled
Author => Phalcon Team and contributors
Version => 5.1.1
Build Date => Nov 14 2022 22:46:48
Powered by Zephir => Version 0.16.3-$Id$
```

**Comment by kgrammer** [comment]
I've always had to install using the compile method because every time I try to install using pecl, I get a lot of undefined errors and it fails to build.

I will reproduce that issue and open a separate ticket as soon as I can get to it.

I am using these command to compile on Ubuntu 22.04:

cd /usr/local/lib
git clone https://github.com/phalcon/cphalcon
cd cphalcon
git checkout tags/v5.1.0 ./
zephir fullclean
zephir build

**Comment by kgrammer** [comment]
And, when I do the "php --ri phalcon" command, I get:

Phalcon is a full stack PHP framework, delivered as a PHP extension, offering lower resource consumption and high performance.
phalcon => enabled
Author => Phalcon Team and contributors
Version => 5.1.0
Build Date => Nov 14 2022 17:09:22
Powered by Zephir => Version 0.16.0-4fac47b

**Comment by Jeckerson** [comment]
> git checkout tags/v5.1.0 ./

Well, here is the answer, change to
```
git checkout tags/v5.1.1 ./
```

**Comment by kgrammer** [comment]
This is strange.  I'm getting different results on different servers.

On server 1, when I ran "git checkout tags/v5.1.1 ./" I received "Updated 0 paths from fb9ce23c4"
I then ran "git checkout tags/v5.1.0 ./" and I got "Updated 18 paths from bdc3f1537"
I followed that with another "git checkout tags/v5.1.1 ./" and got "Updated 18 paths from fb9ce23c4"

On server 2 I ran "git checkout tags/v5.1.1 ./" and got "Updated 185 paths from fb9ce23c4"

I cleared the cphalcon folder on server 1 and did a full get as follows:

root@production:/usr/local/lib# git clone https://github.com/phalcon/cphalcon
Cloning into 'cphalcon'...
remote: Enumerating objects: 252300, done.
remote: Total 252300 (delta 0), reused 0 (delta 0), pack-reused 252300
Receiving objects: 100% (252300/252300), 212.85 MiB | 27.42 MiB/s, done.
Resolving deltas: 100% (201243/201243), done.
root@production:/usr/local/lib# cd cphalcon/
root@production:/usr/local/lib/cphalcon# git checkout tags/v5.1.1 ./
Updated 0 paths from fb9ce23c4

After the build on server 1, it "php --ri phalcon" now shows:

Phalcon is a full stack PHP framework, delivered as a PHP extension, offering lower resource consumption and high performance.
phalcon => enabled
Author => Phalcon Team and contributors
Version => 5.1.1
Build Date => Nov 15 2022 09:04:36
Powered by Zephir => Version 0.16.0-4fac47b

So the build on server 1 seems to have worked this time.

Any clues as to why "git checkout tags/v5.1.1 ./" would fail at random times?

(PS.  I have half a dozen servers, so I will be trying this on other servers in a bit...)

**Comment by kgrammer** [comment]
OK, I see what confused me here...

When you issue the command "git checkout tags/v5.1.1 ./" you get:

Updated 0 paths from fb9ce23c4

That appears to be correct, but since I had been used to seeing the previous commands return some number greater than 0, I ASSUMED it didn't work.  I've updated two servers now and they are showing 5.1.1 as expected.  I will carry on and update this again if needed.

I will add that the one server that reported 185 paths updated is really confusing now...
